tests/unit_tests/utils/test_general.py

```diff
@@ -337,3 +413,258 @@ def test_get_file_age(tmp_test_directory):
+    assert not new_args
+
+    # Test with one argument.
+    args = [list(range(10))]
+    new_args = gen.sort_arrays(*args)
+    assert new_args == [list(range(10))]
+
+    # Test with multiple arguments.
+    args = [list(range(10)), list(range(10, 20))]
+    new_args = gen.sort_arrays(*args)
+    assert new_args == [list(range(10)), list(range(10, 20))]
+
+    # Test with arguments of different lengths.
+    args = [list(range(10)), list(range(5))]
+    new_args = gen.sort_arrays(*args)
+    assert new_args == [list(range(10)), list(range(5))]
+
+    # Test with arguments that are not arrays.
+    args = [1, 2, 3]
+    with pytest.raises(TypeError):
+        gen.sort_arrays(*args)
+
+    # Test with the input array not in the right order.
+    args = [list(reversed(range(10)))]
+    new_args = gen.sort_arrays(*args)
+    assert new_args == [list(range(10))]
+
+
+def test_collect_final_lines(tmp_test_directory) -> None:
+    """
+    Test the collect_final_lines function.
+    """
+
+    # Test with no file.
+    with pytest.raises(FileNotFoundError):
+        gen.collect_final_lines("no_such_file.txt", 10)
+
+    # Test with empty file.
+    file = tmp_test_directory / "empty_file.txt"
+    with open(file, "w"):
+        pass
+    assert gen.collect_final_lines(file, 10) == ""
+
+    # Test with one line file.
+    file = tmp_test_directory / "one_line_file.txt"
+    with open(file, "w") as f:
+        f.write("Line 1")
+    assert gen.collect_final_lines(file, 1) == "Line 1"
+
+    # In the following tests the \n in the output are removed, but in the actual print statements,
+    # where the original function is used, they are still present in the string representation.
+
+    # Test with multiple lines file.
+    file = tmp_test_directory / "multiple_lines_file.txt"
+    with open(file, "w") as f:
+        f.write("Line 1\nLine 2\nLine 3")
+    assert gen.collect_final_lines(file, 2) == "Line 2Line 3"
+
+    # Test with file with n_lines lines.
+    file = tmp_test_directory / "n_lines_file.txt"
+    with open(file, "w") as f:
+        for i in range(10):
+            f.write(f"Line {i}\n")
+        f.write("Line 10")
+    assert gen.collect_final_lines(file, 3) == "Line 8Line 9Line 10"
+
+    # Test with file compressed in gzip.
+    file = tmp_test_directory / "compressed_file.txt.gz"
+    with gzip.open(file, "wb") as f:
+        f.write(b"Line 1\nLine 2\nLine 3")
+    assert gen.collect_final_lines(file, 2) == "Line 2Line 3"
+
+
+def test_log_level_from_user() -> None:
+    """
+    Test get_log_level_from_user() function.
+    """
+    assert gen.get_log_level_from_user("info") == logging.INFO
+    assert gen.get_log_level_from_user("debug") == logging.DEBUG
+    assert gen.get_log_level_from_user("warn") == logging.WARNING
+    assert gen.get_log_level_from_user("warning") == logging.WARNING
+    assert gen.get_log_level_from_user("error") == logging.ERROR
+    assert gen.get_log_level_from_user("critical") == logging.CRITICAL
```

Comment:
We use nowhere in the code critical. Is this necessary?

Reply:
Yes the test is necessary cause it's an available option for logging and we might use it at some point in the future.